sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/CredentialsSupportedSamples.java

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai;
+
+import com.azure.ai.openai.models.Choice;
+import com.azure.ai.openai.models.Completions;
+import com.azure.ai.openai.models.CompletionsOptions;
+import com.azure.ai.openai.models.CompletionsUsage;
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.Configuration;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Sample demonstrates how to create a synchronous client with (1) AzureKeyCredential, (2) Azure Active Directory, and
+ * (3) Public Non-Azure Key.
+ */
+public class CredentialsSupportedSamples {
```

Comment:
We should have separate samples for each credential type. Samples should be runnable and easy to copy-paste. Having all credential types makes it hard to select parts that are needed for one credential type.